torch/_inductor/compile_fx_async.py

```diff
@@ -0,0 +1,169 @@
+        future: Future[_WireProtocolPickledOutput],
+        # this callback gets called to turn the _WireProtocolPickledOutput into an OutputCode
+        callback: Callable[[_WireProtocolPickledOutput], OutputCode],
+    ) -> None:
+        self._eager_forward = eager_forward
+        self._output_code = None
+
+        self._future = future
+        self._callback = callback
+
+        # This tells callers to call us with `inputs` instead of `*inputs`.
+        self._boxed_call = True
+
+    @override
+    def __call__(self, inputs: Sequence[Any]) -> Any:
+        if self._future is not None and self._future.done():
+            self._switch_to_compiled_forward()
+
+        if eager_forward := self._eager_forward:
+            _AsyncFxCompile._stat_eager_runs += 1
+            # TODO: Is this correct? Should eager_forward() always be called as
+            # a boxed call?
+            return eager_forward(inputs)
+
+        else:
+            _AsyncFxCompile._stat_oop_runs += 1
+            assert self._output_code is not None
+            if getattr(self._output_code, "_boxed_call", False):
+                return self._output_code.__call__(inputs)
+            else:
+                return self._output_code.__call__(*inputs)
+
+    def _switch_to_compiled_forward(self) -> None:
+        assert self._future is not None
+
+        # TODO: If the future ended in an exception do we want to continue
+        # running eager or hit the exception now?
+        f, self._future = self._future, None
+        output_code = self._callback(f.result())
+
+        if pcd := self._post_compile_data:
+            self._post_compile_data = None
+            output_code.post_compile(pcd.example_inputs, pcd.cudagraphs, pcd.constants)
+
+        self._output_code = output_code
+        self._eager_forward = None
+
+    @override
+    def post_compile(
+        self,
+        example_inputs: Sequence[InputType],
+        cudagraphs: BoxedBool,
+        constants: CompiledFxGraphConstants,
+    ) -> None:
+        if self._eager_forward is not None:
+            self._post_compile_data = _PostCompileData(
+                example_inputs, cudagraphs, constants
+            )
+        else:
+            assert self._output_code is not None
+            self._output_code.post_compile(example_inputs, cudagraphs, constants)
+
+
+# Given an FxCompile for an out-of-process compile _AsyncFxCompile will run
+# eager until the compiled artifact is ready then it will automatically switch
+# over to using the compiled version.
+@final
+class _AsyncFxCompile(FxCompile):
+    _compile: _OutOfProcessFxCompile
+
+    # Some debugging stats:
+    # Number of times we started a background compile.
+    _stat_bg_started: int = 0
+    # Number of times we finished a background compile.
+    _stat_bg_finished: int = 0
+    # Number of times we ran "eager"
+    _stat_eager_runs: int = 0
+    # Number of times we ran our compiled (out-of-process) artifact
+    _stat_oop_runs: int = 0
```

Comment:
```suggestion
    _stat_compiled_runs: int = 0
```
The name `oop` was confusing me.  I think out-of-process is implied by the class we are in.